### PR TITLE
optimization debian package manager tweaks

### PR DESCRIPTION
**Pull request by Rajpratik71**
## What does this PR (Pull Request) do?
- [x] This PR is not related to any Issue

By default, Ubuntu-based or Debian-based `apt` or `apt-get` system installs recommended - but not suggested - packages .

By passing `--no-install-recommends"\` option, the user lets `apt-get` know not to consider recommended packages as a dependency to install.

This results in smaller downloads and installation of packages .

Refer to blog at [Ubuntu Blog](https://ubuntu.com/blog/we-reduced-our-docker-images-by-60-with-no-install-recommends) .

This PR inserts this option into calls to `apt`/`apt-get` in various places throughout the code base.

## Which Traffic Control components are affected by this PR?
- CDN in a Box
- Documentation

## What is the best way to verify this PR?
Build documentation and build and run affected Docker image specifications.

## The following criteria are ALL met by this PR
- [x] Only affects dependency management; building/running is the only testing that makes sense.
- [x] This PR includes documentation
- [x] An update to CHANGELOG.md is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

File: docs/source/admin/quick_howto/ciab.rst [hunk 1]
```diff
@@ -376,7 +376,7 @@ On Linux, we suggest ``openvpn``. On most Linux distributions, this will also be
 .. code-block:: shell
 	:caption: Install openvpn on ubuntu/debian
 
-	apt-get update && apt-get install -y openvpn
+	apt-get update && apt-get --no-install-recommends install -y openvpn
```
Comment (ocket8888) [inline docs/source/admin/quick_howto/ciab.rst]: This is just for people to install a vpn client on the host, so I don't think there's any need for this.

File: infrastructure/cdn-in-a-box/optional/vpn/README.rst [hunk 2]
```diff
@@ -39,7 +39,7 @@ On Linux, you could choose ``openvpn``. Take ubuntu/debian as an example, you ca
 
 .. code-block:: shell
 
-    apt-get update && apt-get install -y openvpn
+    apt-get update && apt-get --no-install-recommends install -y openvpn
```
Comment (ocket8888) [inline infrastructure/cdn-in-a-box/optional/vpn/README.rst]: Same as above

File: docs/source/admin/quick_howto/ciab.rst (hunk 1 above)
Comment (Rajpratik71) [reply docs/source/admin/quick_howto/ciab.rst]: I general updated "apt-get install" with "apt-get --no-install-recommends install" as a best practice for apt-get wherever applicable

File: infrastructure/cdn-in-a-box/optional/vpn/README.rst (hunk 2 above)
Comment (Rajpratik71) [reply infrastructure/cdn-in-a-box/optional/vpn/README.rst]: I general updated "apt-get install" with "apt-get --no-install-recommends install" as a best practice for apt-get wherever applicable  , doesn't have any issue and didn't looked for specific use case

File: docs/source/admin/quick_howto/ciab.rst (hunk 1 above)
Comment (ocket8888) [reply docs/source/admin/quick_howto/ciab.rst]: Right, but in this case the user is installing something on their own computer. `--no-install-recommends` is a "best practice" for the context of docker, but ostensibly the user has the default apt configuration they want on their own host computer. We don't need to worry - and shouldn't worry - about policing that.

File: docs/source/admin/traffic_vault.rst [hunk 3]
```diff
@@ -136,7 +136,7 @@ On each Traffic Vault server follow these steps.
 		yum install -y java-1.8.0-openjdk java-1.8.0-openjdk-devel
 
 		# On Ubuntu/Debian/Linux Mint
-		apt install -y openjdk-8-jdk
+		apt-get --no-install-recommends install -y openjdk-8-jdk
```
Comment (ocket8888) [inline docs/source/admin/traffic_vault.rst]: Same as above RE: user install recommendations.

File: infrastructure/cdn-in-a-box/dns/Dockerfile [hunk 4]
```diff
@@ -23,10 +23,10 @@ ENV BIND_USER=bind \
 RUN echo 'APT::Install-Recommends 0;' >> /etc/apt/apt.conf.d/01norecommends \
  && echo 'APT::Install-Suggests 0;' >> /etc/apt/apt.conf.d/01norecommends \
  && apt-get update \
- && DEBIAN_FRONTEND=noninteractive apt-get install -y vim.tiny wget net-tools sudo net-tools ca-certificates unzip apt-transport-https \
+ && DEBIAN_FRONTEND=noninteractive apt-get --no-install-recommends install -y vim.tiny wget net-tools sudo net-tools ca-certificates unzip apt-transport-https \
```
Comment (zrhoffman) [inline infrastructure/cdn-in-a-box/dns/Dockerfile:26]: The lines above

```dockerfile
RUN echo 'APT::Install-Recommends 0;' >> /etc/apt/apt.conf.d/01norecommends \
 && echo 'APT::Install-Suggests 0;' >> /etc/apt/apt.conf.d/01norecommends \
```
updating the APT config to skip package recommendations, so there is no need to pass the `--no-install-recommends` option with every apt or apt-get command.

File: infrastructure/cdn-in-a-box/dns/Dockerfile [hunk 5]
```diff
@@ -23,10 +23,10 @@ ENV BIND_USER=bind \
...
  && rm -rf /var/lib/apt/lists/* && rm -rf /etc/apt/apt.conf.d/docker-gzip-indexes \
  && apt-get update \
- && DEBIAN_FRONTEND=noninteractive apt-get install -y bind9=${BIND_VERSION}* bind9-host=${BIND_VERSION}* dnsutils \
+ && DEBIAN_FRONTEND=noninteractive apt-get --no-install-recommends install -y bind9=${BIND_VERSION}* bind9-host=${BIND_VERSION}* dnsutils \
```
Comment (zrhoffman) [inline infrastructure/cdn-in-a-box/dns/Dockerfile:29]: Same as above—We already disable `APT::Install-Recommends` in the APT configuration in this Dockerfile, so no need to pass the `--no-install-recommends` CLI option.

File: infrastructure/cdn-in-a-box/enroller/Dockerfile [hunk 6]
```diff
@@ -31,7 +31,7 @@ RUN go clean && go get -v && go build
 
 FROM debian:stretch
 
-RUN apt-get update && apt-get install -y netcat curl dnsutils net-tools && apt-get clean
+RUN apt-get update && apt-get --no-install-recommends install -y netcat curl dnsutils net-tools && apt-get clean
```
Comment (zrhoffman) [inline infrastructure/cdn-in-a-box/enroller/Dockerfile:34]: See above—I would prefer disabling `APT::Install-Recommends` and `APT::Install-Suggests` in the APT configuration to passing `--no-install-recommends` for every apt or apt-get command.

File: infrastructure/cdn-in-a-box/traffic_vault/Dockerfile [hunk 7]
```diff
@@ -23,7 +23,7 @@ RUN rm -rfv /etc/riak/prestart.d/* /etc/riak/poststart.d/*
 RUN echo 'APT::Install-Recommends 0;' >> /etc/apt/apt.conf.d/01norecommends \
  && echo 'APT::Install-Suggests 0;' >> /etc/apt/apt.conf.d/01norecommends \
  && apt-get update \
- && DEBIAN_FRONTEND=noninteractive apt-get install -y net-tools ca-certificates dnsutils gettext-base \
+ && DEBIAN_FRONTEND=noninteractive apt-get --no-install-recommends install -y net-tools ca-certificates dnsutils gettext-base \
```
Comment (zrhoffman) [inline infrastructure/cdn-in-a-box/traffic_vault/Dockerfile:26]: As with the `dns` Dockerfile,

```dockerfile
RUN echo 'APT::Install-Recommends 0;' >> /etc/apt/apt.conf.d/01norecommends \
 && echo 'APT::Install-Suggests 0;' >> /etc/apt/apt.conf.d/01norecommends \
```
already updates the APT config to skip package recommendations, so there is no need to pass the `--no-install-recommends` option with every apt or apt-get command.

File: traffic_portal/Dockerfile [hunk 8]
```diff
@@ -14,7 +14,7 @@
 FROM node:4-onbuild
 #FROM buildpack-deps:jessie
 
-RUN apt-get update -y && apt-get install libffi-dev ruby-dev rubygems vim -y
+RUN apt-get update -y && apt-get --no-install-recommends install -y libffi-dev ruby-dev rubygems vim -y
```
Comment (zrhoffman) [inline traffic_portal/Dockerfile:17]: See above—I would prefer disabling `APT::Install-Recommends` and `APT::Install-Suggests` in the APT configuration to passing `--no-install-recommends` for every apt or apt-get command.

File: traffic_portal/docker/Dockerfile [hunk 9]
```diff
@@ -14,7 +14,7 @@
 FROM node:4-onbuild
 #FROM buildpack-deps:jessie
 
-RUN apt-get update -y && apt-get install libffi-dev ruby-dev rubygems vim -y
+RUN apt-get update -y && apt-get --no-install-recommends install -y libffi-dev ruby-dev rubygems vim -y
```
Comment (zrhoffman) [inline traffic_portal/docker/Dockerfile:17]: See above—I would prefer disabling `APT::Install-Recommends` and `APT::Install-Suggests` in the APT configuration to passing `--no-install-recommends` for every apt or apt-get command.

File: infrastructure/cdn-in-a-box/traffic_stats/Dockerfile-influxdb [hunk 10]
```diff
@@ -18,7 +18,7 @@
 FROM influxdb:1.7.3
 
 RUN apt-get update && \
-    apt-get install -y dnsutils net-tools gettext-base netcat && \
+    apt-get --no-install-recommends install -y dnsutils net-tools gettext-base netcat && \
```
Comment (zrhoffman) [inline infrastructure/cdn-in-a-box/traffic_stats/Dockerfile-influxdb:21]: See above—I would prefer disabling `APT::Install-Recommends` and `APT::Install-Suggests` in the APT configuration to passing `--no-install-recommends` for every apt or apt-get command.

File: infrastructure/cdn-in-a-box/optional/vpn/Dockerfile [hunk 11]
```diff
@@ -16,7 +16,7 @@
 # under the License.
 
 FROM ubuntu:18.04
-RUN apt-get update && apt-get install -y vim tree wget iptables openvpn iputils-ping net-tools dnsutils
+RUN apt-get update && apt-get --no-install-recommends install -y vim tree wget iptables openvpn iputils-ping net-tools dnsutils
```
Comment (zrhoffman) [inline infrastructure/cdn-in-a-box/optional/vpn/Dockerfile:19]: See above—I would prefer disabling `APT::Install-Recommends` and `APT::Install-Suggests` in the APT configuration to passing `--no-install-recommends` for every apt or apt-get command.

File: infrastructure/cdn-in-a-box/optional/grafana/Dockerfile [hunk 12]
```diff
@@ -21,7 +21,7 @@ USER root
 ARG TRAFFIC_TS_RPM=traffic_stats/traffic_stats.rpm
 
 RUN apt-get update && \
-    apt-get install -y dnsutils net-tools gettext-base p7zip-full netcat && \
+    apt-get --no-install-recommends install -y dnsutils net-tools gettext-base p7zip-full netcat && \
```
Comment (zrhoffman) [inline infrastructure/cdn-in-a-box/optional/grafana/Dockerfile:24]: See above—I would prefer disabling `APT::Install-Recommends` and `APT::Install-Suggests` in the APT configuration to passing `--no-install-recommends` for every apt or apt-get command.

File: infrastructure/cdn-in-a-box/optional/vpn/README.rst (hunk 2 above)
Comment (zrhoffman) [reply infrastructure/cdn-in-a-box/optional/vpn/README.rst:42]: Agree with @ocket8888—We're not in the business of policing a user's own APT configuration.

File: experimental/postgrest/Postgrest_Dockerfile [hunk 13]
```diff
@@ -25,7 +25,7 @@
 FROM ubuntu:15.10
 MAINTAINER dev@trafficcontrol.apache.org
 
-RUN apt-get install -y curl tar xz-utils postgresql-client
+RUN apt-get --no-install-recommends install -y curl tar xz-utils postgresql-client
```
Comment (zrhoffman) [inline experimental/postgrest/Postgrest_Dockerfile:28]: I would prefer disabling `APT::Install-Recommends` and `APT::Install-Suggests` in the APT configuration to passing `--no-install-recommends` for every apt or apt-get command.

**Comment by ocket8888** [comment]
closing as stale with conflicts; please re-base and re-open or open a new PR to get these changes merged.

**Pull request closed without merging**